test/jobs/job-sharing.ts

```diff
@@ -0,0 +1,225 @@
+        it('returns a 200 response', function () {
+          expect(this.res.statusCode).to.equal(200);
+        });
+      });
+    });
+
+    describe('Collection harmony.has-eula tag = true, but guests users have CMR read permissions for the collection', function () {
+      describe('Accessing the job status page', function () {
+        hookJobStatus({ jobID: jobIDWithEULATrueAndGuestReadTrue, username: 'jill' });
+        it('returns a 404 response', function () {
+          expect(this.res.statusCode).to.equal(404);
+        });
+      });
+      describe('Accessing the STAC Catalog page', function () {
+        hookStacCatalog(jobIDWithEULATrueAndGuestReadTrue, 'jill');
+        it('returns a 404 response', function () {
+          expect(this.res.statusCode).to.equal(404);
+        });
+      });
+      describe('Accessing the STAC Item page', function () {
+        hookStacItem(jobIDWithEULATrueAndGuestReadTrue, 0, 'jill');
+        it('returns a 404 response', function () {
+          expect(this.res.statusCode).to.equal(404);
+        });
+      });
+    });
+
+    describe('Collection harmony.has-eula tag = false, but guest users do not have CMR read permissions', function () {
+      describe('Accessing the job status page', function () {
+        hookJobStatus({ jobID: jobIDWithEULAFalseAndGuestReadFalse, username: 'jill' });
+        it('returns a 404 response', function () {
+          expect(this.res.statusCode).to.equal(404);
+        });
+      });
+      describe('Accessing the STAC Catalog page', function () {
+        hookStacCatalog(jobIDWithEULAFalseAndGuestReadFalse, 'jill');
+        it('returns a 404 response', function () {
+          expect(this.res.statusCode).to.equal(404);
+        });
+      });
+      describe('Accessing the STAC Item page', function () {
+        hookStacItem(jobIDWithEULAFalseAndGuestReadFalse, 0, 'jill');
+        it('returns a 404 response', function () {
+          expect(this.res.statusCode).to.equal(404);
+        });
+      });
+    });
+
+    describe('Collection has no harmony.has-eula tag set', function () {
+      describe('Accessing the job status page', function () {
+        hookJobStatus({ jobID: jobIDWithEULANonexistent, username: 'jill' });
+        it('returns a 404 response', function () {
+          expect(this.res.statusCode).to.equal(404);
+        });
+      });
+      describe('Accessing the STAC Catalog page', function () {
+        hookStacCatalog(jobIDWithEULANonexistent, 'jill');
+        it('returns a 404 response', function () {
+          expect(this.res.statusCode).to.equal(404);
+        });
+      });
+      describe('Accessing the STAC Item page', function () {
+        hookStacItem(jobIDWithEULANonexistent, 0, 'jill');
+        it('returns a 404 response', function () {
+          expect(this.res.statusCode).to.equal(404);
+        });
+      });
+    });
+
+    describe('No collections are used in job', function () {
```

Comment:
I guess for backward compatibility (where we haven't saved a value to the collectionIds column when the job was originally kicked off) this should actually return a 404 when someone other than the owner tries to access the job results?

Reply:
https://github.com/nasa/harmony/pull/51/commits/8f3cccdb9a8c05ff14f037af015a93af741279c8

Reply:
I think you are right - we should change it so if we can't tell what the collections for the job are - only the owner has access to it. Hopefully an easy change to use `await Job.byUsernameAndRequestId` in that case instead of `await Job.byRequestId`.